query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/batch/select_one_singular.rs

```diff
@@ -342,9 +342,7 @@ mod singular_batch {
     }
 
     async fn compact_batch(runner: &Runner, queries: Vec<String>) -> TestResult<(QueryResult, BatchDocument)> {
-        // Ensure batched queries are valid
         let res = runner.batch(queries.clone(), false, None).await?;
-        res.assert_success();
```

Comment:
This line is going away because there could be transient errors and we don't want to globally error on these, or because this is downstream of the retry logic?

Reply:
<img width="666" alt="image" src="https://user-images.githubusercontent.com/25233379/211530975-ef721848-4106-454e-a608-272f972048de.png">

For some reason, on batch queries, only top-level errors were checked for the assertions. This prevented the batch query regression test I added to fail.

Now, the tests that use this helper you quoted are _expected_ to yield errors sometimes. Does that make sense?

Reply:
Absolutely! thanks